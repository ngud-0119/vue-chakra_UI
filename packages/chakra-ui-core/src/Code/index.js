import Box from '../Box'
import useBadgeStyle from '../Badge/badge.styles'
import { useVariantColorWarning, forwardProps } from '../utils'
import { baseProps } from '../config/props'

const Code = {
  name: 'Code',
  inject: ['$theme', '$colorMode'],
  props: {
    variantColor: {
      type: String,
      default: 'gray'
    },
    ...baseProps
  },
  computed: {
    theme () {
      return this.$theme()
    },
    colorMode () {
      return this.$colorMode()
    },
    badgeStyle () {
      useVariantColorWarning(this.theme, 'Code', this.variantColor)
      return useBadgeStyle({
        variant: 'subtle',
        color: this.variantColor,
        colorMode: this.colorMode,
        theme: this.theme
      })
    }
  },
  render (h) {
    return h(Box, {
      props: {
        as: 'code',
        display: 'inline-block',
        fontFamily: 'mono',
        fontSize: 'sm',
        px: '0.2em',
        rounded: 'sm',
        ...this.badgeStyle,
        ...forwardProps(this.$props)
      }
    }, this.$slots.default)
  }
}

export default Code
