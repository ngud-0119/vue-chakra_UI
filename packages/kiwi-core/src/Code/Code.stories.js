import { storiesOf } from '@storybook/vue'
import { CSSReset, Code, Stack } from '..'

storiesOf('UI | Code', module)
  .add('Default Code', () => ({
    components: { CSSReset, Code },
    template: `
      <div>
        <CSSReset />
        <Code>Hello world</Code>
      </div>
    `
  }))
  .add('With variant color', () => ({
    components: { CSSReset, Code, Stack },
    template: `
      <div>
        <CSSReset />
        <Stack isInline>
          <Code>console.log(welcome)</Code>
          <Code variantColor="red">var chakra = 'awesome!'</Code>
          <Code variantColor="yellow" >npm install chakra</Code>
        </Stack>
      </div>
    `
  }))
