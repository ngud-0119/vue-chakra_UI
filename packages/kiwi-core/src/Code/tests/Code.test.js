import Code from '../'
import { render } from '@/tests/test-utils'

const renderComponent = (props) => {
  const base = {
    components: { Code },
    template: `<Code>content</Code>`,
    ...props
  }
  return render(base)
}

it('should render correctly', () => {
  const { asFragment } = renderComponent()
  expect(asFragment()).toMatchSnapshot()
})

it('should display children', () => {
  const { getByText } = renderComponent()

  expect(getByText('content')).toBeInTheDocument()
})
