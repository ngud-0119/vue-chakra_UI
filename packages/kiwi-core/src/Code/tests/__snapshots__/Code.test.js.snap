// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<DocumentFragment>
  <code
    class="css-6de9to"
  >
    content
  </code>
</DocumentFragment>
`;
